cloudbuild/presubmit.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+configure_gcloud() {
+  gcloud config set core/disable_prompts TRUE
+  gcloud config set compute/region us-central1
+  gcloud config set compute/zone us-central1-f
+}
+
+install_test_dependencies() {
+  pip3 install -r integration_tests/requirements.txt
+}
+
+initialize_git_repo() {
+  git init
+
+  git remote add origin "https://github.com/GoogleCloudPlatform/dataproc-initialization-actions.git"
+  git fetch origin master
+
+  git reset origin/master
+}
+
+# Determines whether a given string is a prefix string of any changed file name
+is_prefix() {
+  for file in "${CHANGED_FILES[@]}"; do
+    if [[ $file =~ ^$1 ]]; then
+      return 0
+    fi
+  done
+  return 1
+}
+
+determine_tests_to_run() {
```

Comment:
Can we add comments about how we detect changes? seems you cloned a repo, then somehow calculated the diff, but I don't understand.

Reply:
Done.